pkg/servicemesh/controller/controller.go

```diff
@@ -121,19 +107,38 @@ func (smmrc *serviceMeshMemberRollController) newServiceMeshMemberRollListener(l
 		name:              name,
 	}
 
-	if smmrc.cacheWarmed {
-		smmrLog.Debugf("Listener for %q created. Ready to send an initial update", name)
+	// WARNING: This is a bit of a hack.  It is mostly intended as a temporary
+	// workaround to enable 2.1 testing to begin.
+	//
+	// Previously we sent an immediate initial update to all listeners when they
+	// were registered that included only the Istio system namespace.  That
+	// caused problems with some controllers, e.g. IOR, because they expect the
+	// callback to be called with the full authoritative set of namespaces and
+	// may remove resources for namespaces not in the list.
+	//
+	// This instead waits for the informer's cache to sync, then sends an
+	// initial update only if the expected SMMR is not found in the cache.
+	// Unfortunately, there is a race here between sending this initial update
+	// and a user creating an SMMR resource.
+	go func() {
```

Comment:
this will now create one goroutine per listener that'll poll the `HasSynced()` func... not sure if that's ideal. It does avoid having to use a mutex, but I'm not sure about the trade-off there. before the cache is synced, we're now creating ~20 goroutines all hammering the SMMR informer. it's probably not a big deal, but do you think it's better than a single goroutine + mutex?

Reply:
I see what you mean, but I don't expect it to be a problem in practice. It's not that many threads, and they are polling with a reasonable interval, and the `HasSynced()` method on informers is not doing any heavy lifting -- it's not hitting the API or anything. This seemed fine and easier to read to me since it's all in one place.

Reply:
yeah I agree, it's not a ton of goroutines and they really don't do much

Reply:
I prefer the single goroutine+lock here, but since it's a temporary fix, that's ok.

Reply:
This uses a flag and a RW mutex now, though it's not a single go routine. We still have to poll in the listeners.